### DISCUSSION TITLE
Is it possible to `exclude_unset` on a child instance, but *not* on the parent?

### DISCUSSION THREAD

**Discussion by onecrayon** [pydantic/pydantic#7315] [Question]
I have a class setup similar to this:

```python
from pydantic import BaseModel

class HeaderInfo(BaseModel):
    required_prop: int
    optional_prop_1: str | None = None
    optional_prop_2: str | None = None

class MainPayload(BaseModel):
    header: HeaderInfo
    nullable_prop: int | None = None
```

When I export this to JSON, I would *really* like to `exclude_unset` on the `header`, but *not* on the MainPayload (the reason for this is that the header has a LOT of optional properties depending on the specific payload, whereas the `None` values in the main payload have actual meaning). Is there a recommended way to do this? I considered overriding `model_dump_json` on the MainPayload class, but that feels extremely fragile and like a huge headache for Future Me when it inevitably breaks.

**Comment by sydney-runkle** [comment]
Hi @onecrayon,

Thanks for your question. I think our functional serializers will be of great help to you. The `model_serializer` decorator using `wrap` mode conducts the serialization as normal, then filters based on the fields that have been set on your nested model. Here's an example:

```py
from typing import Any, Dict

from pydantic import BaseModel, model_serializer

class HeaderInfo(BaseModel):
    required_prop: int
    optional_prop_1: str | None = None
    optional_prop_2: str | None = None

    @model_serializer(mode='wrap')
    def serialize_model(self, handler) -> dict[str, Any]:
        result = handler(self)
        return {k: v for k, v in result.items() if k in self.model_fields_set}

class MainPayload(BaseModel):
    header: HeaderInfo
    nullable_prop: int | None = None

header_info = HeaderInfo(required_prop=1)
main_payload = MainPayload(header=header_info)

main_payload.model_dump()
#> {'header': {'required_prop': 1}, 'nullable_prop': None}
```

Let me know if you have any questions! Hope this helps. Thanks for your patience!

**Reply by valq7711** [reply]
I think, it is definitely required `exclude_unset` at model and field configuration level

**Reply by saeedesmaili** [reply]
@sydney-runkle Is it possible to achieve something similar to this but for `exclude_none` (to only exclude `None` from the child model)? I couldn't find an equivalent of `model_fields_set` for this case.

**Comment by happyFish** [comment]
I would love for this behavior to be a bit more flexible. Ideally, I could override the model_dump method on my Child model to set exclude_none or exclude_unset to True, then when calling model_dump() on the Parent, without specifying any kwargs, it would properly use the child's model_dump params to exclude fields accordingly, but use the default values on the Parent model and any other models in the hierarchy.
